plugins/sqlcipherplugin/cryptoplugin.cpp

```diff
@@ -20,11 +21,69 @@
 #include <QtCore/QUuid>
 #include <QtCore/QCryptographicHash>
 
+#include <fstream>
+#include <cstdlib>
+
+#include <openssl/rand.h>
+
 void Sailfish::Secrets::Daemon::Plugins::SqlCipherPlugin::init_aes_encryption()
 {
+    // seed the RNG
+    char seed[16] = {0};
+    std::ifstream rand("/dev/urandom");
+    rand.read(seed, 16);
+    rand.close();
+    RAND_add(seed, 16, 1.0);
+
+    // initialise EVP
     osslevp_init();
 }
 
+Sailfish::Crypto::Result
+Sailfish::Secrets::Daemon::Plugins::SqlCipherPlugin::generateRandomData(
+        quint64 callerIdent,
+        const QString &csprngEngineName,
+        quint64 numberBytes,
+        QByteArray *randomData)
+{
+    Q_UNUSED(callerIdent)
+
+    if (csprngEngineName != Sailfish::Crypto::GenerateRandomDataRequest::DefaultCsprngEngineName) {
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::CryptoPluginRandomDataError,
+                                        QLatin1String("The SQLCipher crypto plugin doesn't currently support other RNG engines")); // TODO!
+    }
+
+    if (numberBytes >= 4097 || numberBytes <= 0) {
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::CryptoPluginRandomDataError,
+                                        QLatin1String("The SQLCipher crypto plugin can only generate up to 4096 bytes of random data at a time"));
+    }
+
+    int nbytes = numberBytes;
+    unsigned char buf[4096];
+    if (RAND_bytes(buf, nbytes) != 1) {
+        return Sailfish::Crypto::Result(Sailfish::Crypto::Result::CryptoPluginRandomDataError,
+                                        QLatin1String("The SQLCipher crypto plugin failed to generate the random data"));
+    }
+
+    *randomData = QByteArray((char*)buf, nbytes);
```

Comment:
I'd prefer getting rid of C-style casts by providing the appropriate g++ compiler warning flag.

Reply:
I will use a C++ reinterpret cast instead